static/js/generic_checkbox.html

```diff
@@ -0,0 +1,304 @@
+}
+
+</style>
+
+<head>
+	<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js" type="text/javascript"></script>
+	<script src="http://52.50.241.19/course-content/2017_N/static/js/server_interaction.js" type="text/javascript"></script>
+	<script src="http://d3js.org/d3.v2.min.js?2.10.0"></script>
+	<script>
+
+		function getUrlVars() {
+		  var vars = {};
+		  var url = document.URL;
+		  url = url.split('%24').join('&');
+		  url = url.split('%3D').join('=');
+
+		  var parts = url.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
+			vars[key] = value;
+		  });
+		  return vars;
+		}
+
+		/*
+			This is a global state dictionary. We define this for convenience when working
+			with edX's state-saving paradigm. Any data that is required for operation to
+			seamlessly persist across page refreshes, or that EdX needs to know about, must
+			be placed in here.
+		*/
+		state = {
+			//is the student's answer correct? (possibly a misnomer... eh)
+				//required for grading the problem
+			'correct_answer' : false,
+			//which choice is the answer?
+			'answer' : -1,
+			//list of user inputs
+				//required to repopulate the form so users don't shit bricks
+			'inputs' : [],
+			//attempts at the problem (how many times getGrade() has been called)
+				//required for our adaptive hints
+			'attempts' : 0,
+			//number of choices
+			'choices' : 0
+		};
+
+		//global variable that is set once the form is displayed.
+		//If this is false we can't access the form's document elements,
+		//because they haven't yet been created.
+		displayed = false;
+
+		//url variables
+		vars = {};
+
+		//just stupid rewrapping the url vars into a global variable
+		function getVars() {
+			vars = getUrlVars();
+		}
+
+        //Compose answer strings for form creation
+        function composeAnswerStringList(div) {
+            var answerlist = [];
+            var row;
+            for (row = 0; row < state['choices']; row++) {
+            	var formstring = '';
+                formstring += '<input type="checkbox" name = "ans" id="ans'+row.toString();
+				formstring += '" onkeypress= "setTimeout(check_answers, 250)"/>';
+				if(div.hasAttribute('choice'+row.toString()))
+				{
+					formstring += div.getAttribute('choice'+row.toString()) + '<br>';
+				}
+				else
+					formstring += '<i>no answer</i><br/>';
+				answerlist.push(formstring);
+            }
+            return answerlist;
+        }
+
+        //Shuffle answerlist items in composed form for answers
+        function shuffleString(iterObject) {
+            var steps;
+            var temp = iterObject.slice(0);
+            var cacheString = '';
+            for (steps = iterObject.length; steps > 0; steps--) {
+                var rand = Math.floor(Math.random() * steps);
+                cacheString += temp.splice(rand, 1);
+            }
+            return cacheString;
+        }
+
+		//Creates the form where users input data. Super simple
```

Comment:
It looks like you're mixing tabs and spaces and the indentation is just explodes. Please recheck that you don't mix them and fix indentation.